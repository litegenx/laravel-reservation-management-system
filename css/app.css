.router-link-active {
  cursor: pointer;
}

