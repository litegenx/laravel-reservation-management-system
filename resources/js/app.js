require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
