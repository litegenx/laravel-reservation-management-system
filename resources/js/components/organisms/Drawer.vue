<template>
    <v-navigation-drawer
        v-if="isAuthenticated"
        :value="isOpenDrawer"
        @input="setDrawerOpen"
        color="secondary"
        fixed
        temporary
        dark
    >
        <v-list
            dense
            nav
            class="py-0 my-4"
        >
            <v-list-item two-line>
                <v-list-item-avatar>
                    <img src="https://randomuser.me/api/portraits/men/81.jpg">
                </v-list-item-avatar>
                <v-list-item-content>
                    <v-list-item-title>{{ userName }}</v-list-item-title>
                </v-list-item-content>
            </v-list-item>

            <v-divider />

            <v-list-item
                v-for="item in menu"
                :key="item.title"
                :to="item.to"
                active-class="amber--text"
            >
                <v-list-item-icon>
                    <v-icon>{{ item.icon }}</v-icon>
                </v-list-item-icon>
                <v-list-item-content>
                    <v-list-item-title>{{ $t(item.title) }}</v-list-item-title>
                </v-list-item-content>
            </v-list-item>
            <v-list-item
                @click.stop="logout"
            >
                <v-list-item-icon>
                    <v-icon>mdi-logout</v-icon>
                </v-list-item-icon>
                <v-list-item-content>
                    <v-list-item-title>{{ $t('pages.logout') }}</v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
</template>

<script>
    import { mapGetters, mapActions } from 'vuex';

    export default {
        computed: {
            ...mapGetters({
                menu: 'getMenu',
                isOpenDrawer: 'common/isOpenDrawer',
                isAuthenticated: 'auth/isAuthenticated',
                userName: 'auth/getUserName',
            }),
        },
        methods: {
            ...mapActions({
                setDrawerOpen: 'common/setDrawerOpen',
                logout: 'auth/logout',
            }),
            async logout () {
                this.setDrawerOpen(false);
                await this.$store.dispatch('auth/logout');
            },
        },
    };
</script>
