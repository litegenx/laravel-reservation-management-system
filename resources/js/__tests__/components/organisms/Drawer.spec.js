import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import Drawer from '../../../components/organisms/Drawer';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

describe('Drawer', () => {
    it('should not render drawer if not authenticated', () => {
        const wrapper = mount(app(Drawer, 'inner'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    getMenu: () => [
                        { title: 'test1', icon: 'mdi-view-dashboard', to: '/test1' },
                        { title: 'test2', icon: 'mdi-settings', to: '/test2' },
                    ],
                    'common/isOpenDrawer': () => false,
                    'auth/isAuthenticated': () => false,
                    'auth/getUserName': () => 'テスト',
                },
                actions: {
                    'common/setDrawerOpen': () => {},
                    'auth/logout': () => {},
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('not authenticated');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-navigation-drawer')).toHaveLength(0);
    });

    it('should not render drawer if not opened', () => {
        const wrapper = mount(app(Drawer, 'inner'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    getMenu: () => [
                        { title: 'test1', icon: 'mdi-view-dashboard', to: '/test1' },
                        { title: 'test2', icon: 'mdi-settings', to: '/test2' },
                    ],
                    'common/isOpenDrawer': () => false,
                    'auth/isAuthenticated': () => true,
                    'auth/getUserName': () => 'テスト',
                },
                actions: {
                    'common/setDrawerOpen': () => {},
                    'auth/logout': () => {},
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('not opened');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-navigation-drawer')).toHaveLength(1);
        expect(wrapper.findAll('.v-navigation-drawer.v-navigation-drawer--fixed')).toHaveLength(1);
        expect(wrapper.findAll('.v-navigation-drawer.v-navigation-drawer--close')).toHaveLength(1);
    });

    it('should render drawer', () => {
        const setDrawerOpen = jest.fn();
        const logout = jest.fn();
        const wrapper = mount(app(Drawer, 'inner'), setupLocalVue({
            store: new Vuex.Store({
                getters: {
                    getMenu: () => [
                        { title: 'test1', icon: 'mdi-view-dashboard', to: '/test1' },
                        { title: 'test2', icon: 'mdi-settings', to: '/test2' },
                    ],
                    'common/isOpenDrawer': () => true,
                    'auth/isAuthenticated': () => true,
                    'auth/getUserName': () => 'テスト',
                },
                actions: {
                    'common/setDrawerOpen': setDrawerOpen,
                    'auth/logout': logout,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot('opened');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('.v-navigation-drawer')).toHaveLength(1);
        expect(wrapper.findAll('.v-navigation-drawer.v-navigation-drawer--fixed')).toHaveLength(1);
        expect(wrapper.findAll('.v-navigation-drawer.v-navigation-drawer--close')).toHaveLength(0);
        expect(wrapper.findAll('.v-list-item__title')).toHaveLength(4);
        expect(wrapper.findAll('.v-list-item__title').at(0).text()).toBe('テスト');
        expect(wrapper.findAll('.v-list-item__title').at(1).text()).toBe('test1');
        expect(wrapper.findAll('.v-list-item__title').at(2).text()).toBe('test2');
        expect(wrapper.findAll('.v-list-item__title').at(3).text()).toBe('Logout');

        wrapper.find('#test-contents').trigger('click');
        expect(setDrawerOpen).toBeCalled();

        wrapper.findAll('.v-list-item__title').at(3).trigger('click');
        expect(logout).toBeCalled();
    });
});
