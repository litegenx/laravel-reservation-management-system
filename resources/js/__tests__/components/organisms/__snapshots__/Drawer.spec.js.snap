// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should not render drawer if not authenticated: not authenticated 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <!---->
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Drawer should not render drawer if not opened: not opened 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <aside
          class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--temporary theme--dark secondary"
          style="height: 100%; top: 0px; transform: translateX(-100%); width: 256px;"
        >
          <div
            class="v-navigation-drawer__content"
          >
            <div
              class="v-list py-0 my-4 v-sheet v-sheet--tile theme--dark v-list--dense v-list--nav"
              role="list"
            >
              <div
                class="v-list-item v-list-item--two-line theme--dark"
                role="listitem"
                tabindex="-1"
              >
                <div
                  class="v-avatar v-list-item__avatar"
                  style="height: 40px; min-width: 40px; width: 40px;"
                >
                  <img
                    src="https://randomuser.me/api/portraits/men/81.jpg"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    テスト
                  </div>
                </div>
              </div>
               
              <hr
                aria-orientation="horizontal"
                class="v-divider theme--dark"
                role="separator"
              />
               
              <a
                class="v-list-item v-list-item--link theme--dark"
                href="#/test1"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-view-dashboard theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Test1
                  </div>
                </div>
              </a>
              <a
                class="v-list-item v-list-item--link theme--dark"
                href="#/test2"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-settings theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Test2
                  </div>
                </div>
              </a>
               
              <div
                class="v-list-item v-list-item--link theme--dark"
                role="listitem"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-logout theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Logout
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-navigation-drawer__border"
          />
        </aside>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Drawer should render drawer: opened 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <aside
          class="v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open v-navigation-drawer--temporary theme--dark secondary"
          style="height: 100%; top: 0px; transform: translateX(0%); width: 256px;"
        >
          <div
            class="v-navigation-drawer__content"
          >
            <div
              class="v-list py-0 my-4 v-sheet v-sheet--tile theme--dark v-list--dense v-list--nav"
              role="list"
            >
              <div
                class="v-list-item v-list-item--two-line theme--dark"
                role="listitem"
                tabindex="-1"
              >
                <div
                  class="v-avatar v-list-item__avatar"
                  style="height: 40px; min-width: 40px; width: 40px;"
                >
                  <img
                    src="https://randomuser.me/api/portraits/men/81.jpg"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    テスト
                  </div>
                </div>
              </div>
               
              <hr
                aria-orientation="horizontal"
                class="v-divider theme--dark"
                role="separator"
              />
               
              <a
                class="v-list-item v-list-item--link theme--dark"
                href="#/test1"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-view-dashboard theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Test1
                  </div>
                </div>
              </a>
              <a
                class="v-list-item v-list-item--link theme--dark"
                href="#/test2"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-settings theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Test2
                  </div>
                </div>
              </a>
               
              <div
                class="v-list-item v-list-item--link theme--dark"
                role="listitem"
                tabindex="0"
              >
                <div
                  class="v-list-item__icon"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate mdi mdi-logout theme--dark"
                  />
                </div>
                 
                <div
                  class="v-list-item__content"
                >
                  <div
                    class="v-list-item__title"
                  >
                    Logout
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="v-navigation-drawer__border"
          />
        </aside>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
